### PR TITLE
roachprod: add custom tags for prometheus config

### PR DESCRIPTION
**Pull request by nameisbhaskar**
few of the labels are missing in prometheus with the change from "gce_sd_configs" to "file_sd_configs". This is because these were extracted from the VM metadata which is not available now. So, the same values are set from roachprod now.

Informs: #124319
Epic: none

### CODE REVIEW

File: pkg/roachprod/roachprod.go [hunk 1]
```diff
@@ -808,10 +808,10 @@ func updatePrometheusTargets(ctx context.Context, l *logger.Logger, c *install.S
 					l.Errorf("error getting the port for node %d: %v", index, err)
 					return
 				}
-				nodeInfo := fmt.Sprintf("%s:%d", v.PublicIP, desc.Port)
+				nodeInfo := fmt.Sprintf("%s:%d", v.PrivateIP, desc.Port)
```
Comment (srosenberg) [inline pkg/roachprod/roachprod.go:811]: Good catch! GCE roachprod clusters are reachable to prometheus via private ips, so we save on egress. We might reconsider at a later time in the context of multi-cloud clusters; for now, those are rare.

File: pkg/roachprod/roachprod.go [hunk 2]
```diff
@@ -826,6 +826,32 @@ func updatePrometheusTargets(ctx context.Context, l *logger.Logger, c *install.S
 	}
 }
 
+// regionRegEx is the regex to extract the region label from zone available as vm property
+var regionRegEx = regexp.MustCompile("(^.+[0-9])(-[a-f]$)")
```
Comment (srosenberg) [inline pkg/roachprod/roachprod.go:830]: I had to check, but it does look like some of the GCE regions [1] are _now_ in double digits, e.g., `europe-west12-a` :)

[1] https://cloud.google.com/compute/docs/regions-zones

File: pkg/roachprod/roachprod.go [hunk 3]
```diff
@@ -826,6 +826,32 @@ func updatePrometheusTargets(ctx context.Context, l *logger.Logger, c *install.S
...
+
+// getLabels returns the labels to be populated in the target configuration in prometheus
+func getLabels(v vm.VM) map[string]string {
+	labels := map[string]string{
+		"cluster":  v.Labels["cluster"],
+		"instance": v.ProviderID,
```
Comment (srosenberg) [inline pkg/roachprod/roachprod.go:836]: Methinks this should be `v.Name`.

File: pkg/roachprod/roachprod.go [hunk 4]
```diff
@@ -826,6 +826,32 @@ func updatePrometheusTargets(ctx context.Context, l *logger.Logger, c *install.S
...
+		"host_ip":  v.PrivateIP,
+		"project":  v.Project,
+		"zone":     v.Zone,
```
Comment (srosenberg) [inline pkg/roachprod/roachprod.go:839]: Missing `region`.

**Review by srosenberg — APPROVED** [review]

File: pkg/roachprod/roachprod.go (hunk 2 above)
Comment (nameisbhaskar) [reply pkg/roachprod/roachprod.go:830]: actually, .* is absorbing the digit as well in this case. I will make it to [0-9]+ to be safer.

Comment (nameisbhaskar) [reply pkg/roachprod/roachprod.go:830]: Thanks for pointing this out!

File: pkg/roachprod/roachprod.go (hunk 3 above)
Comment (nameisbhaskar) [reply pkg/roachprod/roachprod.go:836]: will try the Name and confirm

File: pkg/roachprod/roachprod.go (hunk 4 above)
Comment (nameisbhaskar) [reply pkg/roachprod/roachprod.go:839]: region is in the next lines.

Comment (srosenberg) [reply pkg/roachprod/roachprod.go:839]: 🤦

File: pkg/roachprod/roachprod.go (hunk 3 above)
Comment (nameisbhaskar) [reply pkg/roachprod/roachprod.go:836]: works!! Thanks!

**Comment by nameisbhaskar** [comment]
Thanks @srosenberg for reviewing this!!

**Comment by nameisbhaskar** [comment]
bors r=@srosenberg

**Pull request merged**